File: Data Structures & Algorithms/majority-element/submission-2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = {}
        for n in nums:
            count[n] = 0 if n not in count else count[n]+1
        
        for key in count:
            if count[key] >= len(nums)//2:
                return key
        
        return -1
        

File: Data Structures & Algorithms/two-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const targetMap = new Map();

        for (const i in nums) {
            const cur = target - nums[i];
            if (targetMap.has(cur)) {
                return [Number(i), targetMap.get(cur)];
            } else {
                targetMap.set(Number(nums[i]), Number(i));
            }
        }
        return [-1, -1];

    }
}
